modules/core/shared/src/main/scala/lucuma/core/geom/f2/F2ScienceAreaGeometry.scala

```diff
@@ -0,0 +1,102 @@
+import lucuma.core.math.Angle
+import lucuma.core.math.Offset
+import lucuma.core.math.units.*
+import spire.math.*
+import spire.std.bigDecimal.*
+
+/**
+  * F2 science area geometry.
+  */
+trait F2ScienceAreaGeometry:
+
+  // base target
+  def base: ShapeExpression =
+    ShapeExpression.point(Offset.Zero)
+
+  def scienceAreaDimensions(lyotWheel: F2LyotWheel, fpu: Option[F2Fpu]): (Angle, Angle) =
+    lyotWheel match
+      case F2LyotWheel.F16 | F2LyotWheel.F32High | F2LyotWheel.F32Low =>
+        val pixelScale = lyotWheel.pixelScale
+        val plateScale = BigDecimal(lyotWheel.plateScale).withUnit[ArcSecondPerMillimeter]
+        fpu match
+          case None | Some(F2Fpu.Pinhole) | Some(F2Fpu.SubPixPinhole) =>
+            val size = ImagingFOVSize ⨱ plateScale
+            (size.toAngle, size.toAngle)
+          case Some(fpu)                                              =>
+            (Angle.fromBigDecimalArcseconds(fpu.slitWidth * pixelScale),
+              (LongSlitFOVHeight ⨱ plateScale).toAngle)
+      case _                                                          =>
+        (Angle.Angle0, Angle.Angle0)
+
+  def shapeAt(
+    posAngle:  Angle,
+    offsetPos: Offset,
+    lyotWheel: F2LyotWheel,
+    fpu:       Option[F2Fpu]
+  ): ShapeExpression =
+    val plateScale = BigDecimal(lyotWheel.plateScale).withUnit[ArcSecondPerMillimeter]
+    val scienceAreaWidth = scienceAreaDimensions(lyotWheel, fpu)._1
+    val shape = fpu match
+      case None      => imaging(plateScale)
+      // case _    => mosFOV(plateScale) // TODO: handle MOS
+      case Some(fpu) => longslit(plateScale, scienceAreaWidth)
+    shape ↗ offsetPos ⟲ posAngle
+
+  /**
+   * Create the F2 imaging field of view.
+   * @param plateScale the plate scale in arcsec/mm
+   * @return           a shape representing the FOV
+   */
+  private def imaging(plateScale: F2PlateScale): ShapeExpression =
+    val size = ImagingFOVSize ⨱ plateScale
+    ShapeExpression.centeredEllipse(size.toAngle, size.toAngle)
+
+  /**
+   * Create the F2 MOS field of view shape.
+   * @param plateScale the plate scale in arcsec/mm
+   * @return           a shape representing the FOV
+   */
+  // def mos(plateScale: F2PlateScale): ShapeExpression = {
+  //   // TODO Implement, the logic comes from the ocs but there is no current way to set MOS in the model
+  //   https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/flamingos2/F2ScienceAreaGeometry.scala#L67
+  //   val width  = MOSFOVWidth * plateScale
+  //   val height = ImagingFOVSize * plateScale
+  //   val radius = height / 2.0
+  //
+  //   // The FOV is the intersection of a rectangle and a circle.
+  //   val circle = ShapeExpression.centeredEllipse(-radius.toAngle, height.toAngle)
+  //   val rectangle = ShapeExpression.rectangleAt((-width.toAngle.bisect.p, -radius.toAngle.q), (width.toAngle.p, height.toAngle.q))
+  //   circle ∩ rectangle
+  // }
```

Comment:
Saving some fun for later?

Reply:
Under some definition of fun. There is no MOS on the enums, maybe it is better to add it now and get the geometry working while it is fresh

I'll do that in a next PR